libraries/Microsoft.Bot.Builder/ChannelServiceHandlerBase.cs

```diff
@@ -0,0 +1,490 @@
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// SendConversationHistory() API for Skill.
+        /// </summary>
+        /// <remarks>
+        /// Override this method to this method allows you to upload the historic activities to the conversation.
+        ///
+        /// Sender must ensure that the historic activities have unique ids and
+        /// appropriate timestamps. The ids are used by the client to deal with
+        /// duplicate activities and the timestamps are used by the client to render
+        /// the activities in the right order.
+        /// </remarks>
+        /// <param name="claimsIdentity">claimsIdentity for the bot, should have AudienceClaim, AppIdClaim and ServiceUrlClaim.</param>
+        /// <param name='conversationId'>Conversation ID.</param>
+        /// <param name='transcript'>Transcript of activities.</param>
+        /// <param name='cancellationToken'>The cancellation token.</param>
+        /// <returns>task for a resource response.</returns>
+        protected virtual Task<ResourceResponse> OnSendConversationHistoryAsync(ClaimsIdentity claimsIdentity, string conversationId, Transcript transcript, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// UploadAttachment() API.
+        /// </summary>
+        /// <remarks>
+        /// 
+        /// Override this method to store data in a compliant store when dealing with enterprises.
+        /// 
+        /// The response is a ResourceResponse which contains an AttachmentId which is
+        /// suitable for using with the attachments API.
+        /// </remarks>
+        /// <param name="claimsIdentity">claimsIdentity for the bot, should have AudienceClaim, AppIdClaim and ServiceUrlClaim.</param>
+        /// <param name='conversationId'>Conversation ID.</param>
+        /// <param name='attachmentUpload'>Attachment data.</param>
+        /// <param name='cancellationToken'>The cancellation token.</param>
+        /// <returns>task with result.</returns>
+        protected virtual Task<ResourceResponse> OnUploadAttachmentAsync(ClaimsIdentity claimsIdentity, string conversationId, AttachmentData attachmentUpload, CancellationToken cancellationToken = default)
+        {
+            throw new NotImplementedException();
+        }
+
+        /// <summary>
+        /// Helper to authenticate the header token and extract the claims.
+        /// </summary>
+        /// <param name="authHeader">The auth header containing JWT token.</param>
+        /// <param name="cancellationToken">A cancellation token.</param>
+        /// <returns>A <see cref="ClaimsIdentity"/> representing the claims associated with given header.</returns>
+        protected abstract Task<ClaimsIdentity> AuthenticateAsync(string authHeader, CancellationToken cancellationToken);
```

Comment:
Can we make this internal (and also the overrides in derived classes). We deliberately made this method private in the original ChannelServiceHandler to prevent people from overriding authentication. 

```suggestion
        internal abstract Task<ClaimsIdentity> AuthenticateAsync(string authHeader, CancellationToken cancellationToken);
```